packages/gatsby/src/bootstrap/resolve-module-exports.js

```diff
@@ -54,14 +54,26 @@ const staticallyAnalyzeExports = (modulePath, resolver = require.resolve) => {
       isES6 = true
     },
 
-    // get foo from `export const foo = bar`
     ExportNamedDeclaration: function ExportNamedDeclaration(astPath) {
-      const exportName = get(
-        astPath,
-        `node.declaration.declarations[0].id.name`
-      )
-      isES6 = true
-      if (exportName) exportNames.push(exportName)
+      const node = astPath.node.declaration
+
+      // get foo from `export const foo = bar`
+      if (
+        get(astPath, `node.declaration.type`) === `VariableDeclaration` &&
+        get(astPath, `node.declaration.declarations[0].id.name`)
+      ) {
+        isES6 = true
+        exportNames.push(node.declarations[0].id.name)
+      }
+
+      // get foo from `export function foo()`
+      if (
+        get(astPath, `node.declaration.type`) === `FunctionDeclaration` &&
+        get(astPath, `node.declaration.id.name`)
+      ) {
+        isES6 = true
+        exportNames.push(node.id.name)
```

Comment:
Shouldn't this be
```suggestion
        exportNames.push(node.declaration.id.name)
```
?

Reply:
No, because it`s `const node = astPath.node.declaration`.

But I could rename it to `declaration` :D

Reply:
🤦 ops, my bad - but for my defence naming was confusing xD